src/core/collect/collector.sol

```diff
@@ -34,44 +26,62 @@ contract PileLike {
     function recovery(uint loan, address usr, uint wad) public;
 }
 
-contract Collector {
+contract RegistryLike {
+    function get(uint) public returns (uint);
+}
+
+contract ShelfLike {
+    function claim(uint, address) public;
+    function lock(uint, uint) public;
+}
 
+contract Collector {
     // --- Auth ---
     mapping (address => uint) public wards;
     function rely(address usr) public auth { wards[usr] = 1; }
     function deny(address usr) public auth { wards[usr] = 0; }
     modifier auth { require(wards[msg.sender] == 1); _; }
 
-    SpotterLike spotter;
-    TagLike tag;
+    // --- Data ---
+    RegistryLike liquidation;
+    struct Lot {
+        address usr;
+        uint    wad;
+    }
+    mapping (uint => Lot) public tags;
+
     DeskLike desk;
     PileLike pile;
 
-    constructor (address spotter_, address tag_, address desk_, address pile_) public {
-        spotter = SpotterLike(spotter_);
-        tag = TagLike(tag_);
+    constructor (address desk_, address pile_) public {
         desk = DeskLike(desk_);
         pile = PileLike(pile_);
         wards[msg.sender] = 1;
     }
 
     function depend(bytes32 what, address addr) public auth {
-        if (what == "spotter") spotter = SpotterLike(addr);
-        else if (what == "tag") tag = TagLike(addr);
-        else if (what == "desk") desk = DeskLike(desk);
-        else if (what == "pile") pile = PileLike(pile);
+        if (what == "desk") desk = DeskLike(addr);
+        else if (what == "shelf") shelf = ShelfLike(addr);
+        else if (what == "pile") pile = PileLike(addr);
+        else if (what == "liquidation") liquidation = RegistryLike(addr);
         else revert();
     }
 
-    function collect(uint loan, address usr) public auth  {
-        if(spotter.collectable(loan) == false){
-            spotter.seizure(loan);
-        }
+    // --- Collector ---
+    function file(uint loan, address usr, uint wad) auth {
```

Comment:
```suggestion
    function file(uint loan, address usr, uint wad) public auth {
```

Reply:
For a later auction a `usr` is required but how would it work if we don't care which of the whitelisted address would buy th nft?

The external contract needs to be called to pass my address here?

Reply:
we could add an edge case where if wad > 0 and usr == 0 then any usr could buy it?